src/firefly/js/util/VOAnalyzer.js

```diff
@@ -1247,6 +1247,7 @@ export function applyTokenSub(tableModel, val='', rowIdx, def, encode=false) {
     const vars = val?.match?.(/\${[\w -.]+}/g);
     let rval = val;
     if (vars) {
+        encode = encode && (vars[0] !== val);
```

Comment:
`applyTokenSub` is generic and used in many places.  Please move logic into `applyLinkSub` where it's made for resolving links.  I am guessing something like this..

```
const encode = !!href && !href.match(/^\${[\w -.]+}$/g);    // don't encode if href is blank or consist of exactly one token
if (encode) fval = encodeURIComponent(fval);                // if encoding is needed, then fval also need to be encoded.
const rhref = applyTokenSub(tableModel, href, rowIdx, '', encode);

```

Reply:
Yes I agree it is better way to have the change in 'applyLinkSub' and I was doing that originally.  
I moved the change into 'applyTokenSub'  since I thought, 
- It seems only applyLinkSub has passed 'encode: true' to 'applyTokenSub', other calls pass 'encode:false' to `applyLinkSub`.  (It may not be always true), 
- href.match is done twice in both `applyLinkSub` and `applyTokenSub`, but it should be ok. 

I will move the change to be inside `applyLinkSub` as your comment, however, please check if the match like this
`href.match(/^\${[\w -.]+}$/g)`  is good for the case like `${col_name1}${col_name2}`?

Reply:
Yes,  `/^\${[\w -.]+}$/` is good to test if `href` consists of exact one token! It is for the string starting with `${`, ending with `}`,  and containing at least one of alphanumeric characters, underscore, or characters between space and period in the middle.